src/formatting_stack/strategies.clj

```diff
@@ -80,11 +81,33 @@
        (impl/extract-clj-files)
        (into files)))
 
+(defn current-branch-name []
+  (-> (sh "git" "rev-parse" "--abbrev-ref" "HEAD")
+      (:out)
+      (string/split-lines)
+      (first)))
+
+(defn default-branch-name []
```

Comment:
The introduction of this defn is kinda unrelated, but also necessary for this feature to work.

Anyway, the hardcoded `"master"` was very far from ideal, in fact f-s itself uses `main` as its default branch so it was pretty broken.

The new impl checks for `["master" "main" "stable" "dev"]`. One could also check `origin`, but IME git can opaquely hang over misc network issues. So having f-s hang would be a pain.

This defn could be improved in a more generalized direction in a future, but for now we can observe that it's better than before.

Reply:
I suppose it could also check a system property before making the assumption. If you also _set_ the system-property after it's cached and visible too

Reply:
IDK, instructing people to set a sysprop seems far from ideal.

In a future we can simply refine this defn. It might be possible to do the op offline (remote info is stored locally, except in CI)

Reply:
> This way your addition would work exactly in the same way, but there's an escape hatch to configure this.

Alrighty!